components/icon/icon.directive.ts

```diff
@@ -135,6 +137,14 @@ export class NzIconDirective extends IconDirective implements OnInit, OnChanges,
   private changeIcon2(): void {
     this.setClassName();
 
+    // First of all, we can't render the SVG on the server and send it to the browser for now,
+    // so the browser doesn't have to re-render it. Second, the `IconDirective` will still re-load
+    // this SVG, so we'll have an unnecessary HTTP request on the server-side. There's no sense to
+    // call `_changeIcon()` when running the code on the Node.js side.
+    if (!this.platform.isBrowser) {
+      return;
```

Comment:
Should support changing icon when using static loading. I would provide an API in @ant-design/icons-angular to turn dynamic loading off.

Reply:
Alright, I'll make the changes after that.